lib/client.ts

```diff
@@ -97,8 +97,8 @@ export class Client {
   public patch = (endpoint: Endpoint, payload: any): Promise<any> =>
     this.authorize(this.request(this.url(endpoint), { method: 'PATCH', body: JSON.stringify(payload)}))
 
-  public destroy = (endpoint: Endpoint): Promise<any> =>
-    this.authorize(this.request(this.url(endpoint), { method: 'DELETE' }))
+  public destroy = (endpoint: Endpoint, payload: any): Promise<any> =>
+    this.authorize(this.request(this.url(endpoint), { method: 'DELETE', body: JSON.stringify(payload)}))
```

Comment:
Looks good to me. Is everything fine when no payload is included? 
If you have a moment can you add a spec for delete?
https://github.com/procore/js-sdk/blob/master/test/client.spec.ts